src/Riok.Mapperly/Descriptors/MappingBuilders/DictionaryMappingBuilder.cs

```diff
@@ -131,4 +134,25 @@ private static (ITypeSymbol, ITypeSymbol)? GetEnumerableKeyValueTypes(MappingBui
 
         return (enumeratedType.TypeArguments[0], enumeratedType.TypeArguments[1]);
     }
+
+    private static INamedTypeSymbol? GetExplicitIndexer(MappingBuilderContext ctx)
+    {
+        if (SymbolEqualityComparer.Default.Equals(ctx.Target.OriginalDefinition, ctx.Types.IDictionaryT))
```

Comment:
Wouldn't it make sense to create an overload for the existing `SymbolExtensions.ImplementsGeneric` method with the following signature?
```c#
internal static bool ImplementsGeneric(
        this ITypeSymbol t,
        INamedTypeSymbol genericInterfaceSymbol,
        string symbolName,
        [NotNullWhen(true)] out INamedTypeSymbol? genericIntf,
        out bool isExplicit)
```

The same could then be used for the enumerables and replace `HasImplicitInterfaceMethod` in the end.